### PR TITLE
Feature/extension manager

### PR DESCRIPTION
**Pull request by llali**
- Enabled the extension manger
- Added a static file for extension manager gallery 
- Setting the gallery url in product.json
- Added a new feature to extension manager so extensions can provide a url to browse to when installing the extension in SQL ops instead of installing a VSIX

### CODE REVIEW

File: extensionsGallery.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+"results": [{
+	"extensions": [
+		{
+			"extensionId": "1",
+			"extensionName": "TestExtension",
+			"disolayName": "Test Extension",
```
Comment (MattIrv) [inline extensionsGallery.json]: typo: "displayName"

File: extensionsGallery.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	],
+	"resultMetadata": [{
+		"metadataType": "ResultCount",
+		"metadataItems": [{
+			"name": "TotalCount",
+			"count": 1
+		}]
+	}]
+}]
+}
```
Comment (MattIrv) [inline extensionsGallery.json]: Where does it get all the info that is displayed after you click on the extension? Things like rating, download count, readme, etc. We won't have to maintain that info in this json file will we?

File: product.json [hunk 3]
```diff
@@ -28,5 +28,11 @@
 	"releaseNotesUrl": "https://go.microsoft.com/fwlink/?linkid=862039",
 	"documentationUrl": "https://go.microsoft.com/fwlink/?linkid=862277",
 	"commit": "9ca6200018fc206d67a47229f991901a8a453781",
-	"date": "2017-12-15T12:00:00.000Z"
+	"date": "2017-12-15T12:00:00.000Z",
+	"extensionsGallery": {
+		"serviceUrl":"https://raw.githubusercontent.com/Microsoft/sqlopsstudio/feature/extensionManager/extensionsGallery.json"
```
Comment (MattIrv) [inline product.json]: We'll want to update this to use the master branch

File: src/vs/platform/extensionManagement/node/extensionGalleryService.ts [hunk 4]
```diff
@@ -204,6 +206,13 @@ function getStatistic(statistics: IRawGalleryExtensionStatistics[], name: string
 function getVersionAsset(version: IRawGalleryExtensionVersion, type: string): IGalleryExtensionAsset {
 	const result = version.files.filter(f => f.assetType === type)[0];
 
+	// {{SQL CARBON EDIT}}
+	let properties = version.properties.filter(p => p.key === type);
+	let uriFromProperty: string = undefined;
+	if (properties && properties.length >= 1) {
+		uriFromProperty = properties[0].value;
```
Comment (MattIrv) [inline src/vs/platform/extensionManagement/node/extensionGalleryService.ts]: Looks like this works but I'm not sure I understand how. Why is the uri always `properties[0]`?

File: product.json (hunk 3 above)
Comment (llali) [reply product.json]: this is for testing. I'll change it later to the correct path

---
In reply to: [168322398](https://github.com/Microsoft/sqlopsstudio/pull/689#discussion_r168322398) [](ancestors = 168322398)

File: product.json [hunk 5]
```diff
@@ -28,5 +28,11 @@
...
+	},
+	"extensionTips": {
+		"TestExtension": "{**/*.*}"
```
Comment (llali) [inline product.json]: will remove this. added to verify recommended extensions works

File: src/vs/platform/extensionManagement/node/extensionGalleryService.ts (hunk 4 above)
Comment (llali) [reply src/vs/platform/extensionManagement/node/extensionGalleryService.ts]: technically we can set the property with key  "Microsoft.SQLOps.DownloadPage" more than once but I assume here if it's set, only the first value is valid and it's the uri

---
In reply to: [168323032](https://github.com/Microsoft/sqlopsstudio/pull/689#discussion_r168323032) [](ancestors = 168323032)

File: extensionsGallery.json (hunk 2 above)
Comment (llali) [reply extensionsGallery.json]: I have to remove the rating and download counts from the UI. other information should be from this file. I'll verify that

---
In reply to: [168322330](https://github.com/Microsoft/sqlopsstudio/pull/689#discussion_r168322330) [](ancestors = 168322330)

**Pull request merged**